cmd/transporter/javascript_builder.go

```diff
@@ -295,14 +301,35 @@ func (js *JavascriptBuilder) Build() error {
 
 // Run runs each of the transporter pipelines sequentially
 func (js *JavascriptBuilder) Run() error {
+	var g group.Group
 	for _, p := range js.pipelines {
-		err := p.Run()
-		if err != nil {
-			return err
-		}
+		g.Add(func() error {
+			return p.Run()
+		}, func(error) {
+			p.Stop()
```

Comment:
In the same vein, I was thinking we could start using contexts (with cancel) to coordinate some of this starting/stopping.